spec/requests/rack_attack_spec.rb

```diff
@@ -163,6 +163,39 @@
         expect(analytics).
           to have_received(:track_event).with('Rate Limit Triggered', type: 'req/ip')
       end
+
+      it 'logs the service provider' do
+        analytics = FakeAnalytics.new
+        allow(Analytics).to receive(:new).and_return(analytics)
```

Comment:
I wish there was a cleaner way to do this with like `stub_const` but there really isn't :/